x/mint/types/minter.go

```diff
@@ -40,6 +42,27 @@ func ValidateMinter(minter Minter) error {
 	return nil
 }
 
+// GetInflationWithTime returns the new inflation rate with given time
+func GetInflationWithTime(params Params, genesisTime, currentTime time.Time, initialInflation sdk.Dec,
```

Comment:
we want this to adjust according to bonded ratio as well.

Say the "ideal" inflation, ignoring everything else, is `I`.  we can define a range above and below `I` by a parameter - say `5%`.  Inflation can then be within `[i-.05*i,i+0.05*i]`.  The current inflation can then be adjusted by target bonded ratio, bonded ratio, max inflation change rate, target inflation, and max/min inflation bounds by something like:

* `target_bond_ratio - current_bond_ratio` - determines initial rate of change (including direction)
* `max_inflation_change_rate` - puts an upper bound on the absolute rate of change
* max, min inflation as a function of target (above) - place bounds on final rate of change.

Does that make sense?

Reply:
Sorry @boz , I didn't get this completely.

Reply:
@boz  should we consider max_inflation_change_rate per year? or per block? or some custom period?